ext/gmp/gmp.c

```diff
@@ -1278,12 +1278,37 @@ ZEND_FUNCTION(gmp_pow)
 
 	if (Z_TYPE_P(base_arg) == IS_LONG && Z_LVAL_P(base_arg) >= 0) {
 		INIT_GMP_RETVAL(gmpnum_result);
-		mpz_ui_pow_ui(gmpnum_result, Z_LVAL_P(base_arg), exp);
+		if (exp >= INT_MAX) {
+			mpz_t base_num, exp_num, mod;
+			mpz_init(base_num);
+			mpz_init(exp_num);
+			mpz_init(mod);
+			mpz_set_si(base_num, Z_LVAL_P(base_arg));
+			mpz_set_si(exp_num, exp);
+			mpz_set_ui(mod, UINT_MAX);
```

Comment:
Why compute the power modulo UINT_MAX ?

Reply:
seems large enough and works for both 32/64. What do you have in mind ?

Reply:
I just don't understand why you need to take a modulo. Would this not give the wrong result for e.g. base=2 exp=INT_MAX ?

Reply:
It is more to put a "threshold", originally I implemented an exception.

Reply:
My point is that
```php
var_dump(gmp_pow(2, 2**31));
```

now outputs
```
object(GMP)#1 (1) {
  ["num"]=>
  string(1) "1"
}
```

which is wrong.

I rather have an exception than a wrong result.

Reply:
I know it s wrong :) @girgiad what do you think if I go back to my original idea ?

Reply:
Right, mpz does not have a `mpz_pow` function, which is kinda strange. Throwing an exception is indeed better.